Cython/Compiler/ExprNodes.py

```diff
@@ -2074,7 +2074,11 @@ def calculate_target_results(self, env):
 
     def check_const(self):
         entry = self.entry
-        if entry is not None and not (entry.is_const or entry.is_cfunction or entry.is_builtin):
+        if entry is not None and not (
+                entry.is_const or
+                entry.is_cfunction or
+                entry.is_builtin or
+                entry.type.is_const):
```

Comment:
Would be nice to also have a test case for this change.

Reply:
There is an existing test case in `tests/run/carrays.pyx`. In fact, I made this change because that test case broke.